### DISCUSSION TITLE
2022-11-07 Pyaerocom Meeting

### DISCUSSION THREAD

**Discussion by lewisblake** [metno/pyaerocom#782] [Development meetings]
## Last meeting

#765

## PRs 
- No open PRs :smiley_cat: 

## Other issues
- Conda release of v0.13.2
- #748
-  #671
- #663
- #762
- #770 
- [Pyaerocom development 22-23 ](https://github.com/metno/pyaerocom/projects/9)

**Comment by avaldebe** [comment]
## Conda release of v0.13.2

@avaldebe conda-forge/pyaerocom-feedstock#21 PR is failing to build.
A similar PR created by a bot conda-forge/pyaerocom-feedstock#22 also fails to build.

From the [pipeline logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=605518&view=logs&j=516f237f-98f1-50a4-0946-c43fe5437b81&t=68e286c4-c87b-5a2e-10c4-f21f8bc51a62&l=178
) it seems that `conda_build` does not understand `PEP 517` and needs `flit_core` to be installed into the build environment.

```
ModuleNotFoundError: No module named 'flit_core'
```

**Reply by avaldebe** [reply]
after adding `flit-core` to the host (build) requirements, `conda_build` manages to create the package. However the test fail 

```
tests/test__concprcp_units_helpers.py ...                                [  0%]
tests/test__logging.py ..............................                    [  1%]
tests/test__lowlevel_helpers.py ........................................ [  4%]
..                                                                       [  4%]
tests/test_aux_var_helpers.py .......................                    [  5%]
tests/test_cli.py ......                                                 [  6%]
tests/test_colocateddata.py ............................................ [  8%]
Fatal Python error: Segmentation fault
```

see [pipeline logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=607601&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=356) for more details

**Reply by lewisblake** [reply]
Thanks for checking. Interesting. It seems to be seg-faulting in the tests/fixtures/collocated_data.py. Maybe something (e.g., accessing) with the DataForTests?

**Reply by avaldebe** [reply]
> Thanks for checking. Interesting. It seems to be seg-faulting in the tests/fixtures/collocated_data.py. Maybe something (e.g., accessing) with the DataForTests?

Managed to reproduce the error on my local machine, the data access does not seems to be the problem.

**Reply by avaldebe** [reply]
the problem seems to be `xarray 2022.11.0`, which was released 3 days ago.

**Reply by avaldebe** [reply]
The tests now are failing after `DeprecationWarning` triggered by `importlib.resources` on Python 3.11.

```
tests/io/test_read_eea_aqerep_v2.py::test_read
  $PREFIX/lib/python3.11/site-packages/pyaerocom/io/helpers.py:301: DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
    with resources.path("pyaerocom.data", COUNTRY_CODE_FILE) as path:
```

see [pipeline logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=607717&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=1023) for more details


I will silence the warnings on `conda-forge` tests and further investigate the Python 3.11 `DeprecationWarning` on PR #783.

**Reply by avaldebe** [reply]
the error now is

```
    def test_EbasNasaAmesFile_head_fix_error(filedata):
        with pytest.raises(AttributeError) as e:
            filedata.head_fix = "Blaaaaaaaaaaaaaaa"
>       assert str(e.value).startswith("can't set attribute")
E       assert False
E        +  where False = <built-in method startswith of str object at 0x7f6d60eb1300>("can't set attribute")
E        +    where <built-in method startswith of str object at 0x7f6d60eb1300> = "property 'head_fix' of 'EbasNasaAmesFile' object has no setter".startswith
E        +      where "property 'head_fix' of 'EbasNasaAmesFile' object has no setter" = str(AttributeError("property 'head_fix' of 'EbasNasaAmesFile' object has no setter"))
E        +        where AttributeError("property 'head_fix' of 'EbasNasaAmesFile' object has no setter") = <ExceptionInfo AttributeError("property 'head_fix' of 'EbasNasaAmesFile' object has no setter") tblen=1>.value
```

see [pipeline logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=607733&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=465)


Looks like the problem is related to running under Python 3.11, which we do not fully support.

```
    python:               3.11.0-ha86cf86_0_cpython     conda-forge
```

What would you say about excluding Python 3.11 from this particular release?

**Reply by lewisblake** [reply]
To me that's fair. Thoughts @dulte @jgriesfeller ?

**Reply by jgriesfeller** [reply]
Agreed. Let's exclude Python 3.11 for now.

**Reply by avaldebe** [reply]
conda-forge/pyaerocom-feedstock#21 is passing now (for `python >=3.8,<3.11`), it is now up to @dulte to approve that PR

**Comment by lewisblake** [comment]
Next meeting suggestions: #519 and updating some of the labels for issues (i.e., consider having project-specific labels)
